src/main/java/net/sourceforge/jwbf/mediawiki/actions/queries/WatchList.java

```diff
@@ -0,0 +1,238 @@
+		RequestBuilder requestBuilder = new ApiRequestBuilder().action("query") //
+				.formatJson() //
+				.param("continue", MediaWiki.urlEncode("-||")) //
+				.param("list", "watchlist");
+		requestBuilder.param("wlnamespace",
+				MediaWiki.urlEncodedNamespace(namespaces)).param(
+				"wlprop",
+				joinParam(new HashSet<WatchListProperties>(Arrays
+						.asList(properties))));
+		if (showBots)
+			requestBuilder.param("wlshow", "bot");
+		else
+			requestBuilder.param("wlshow", "!bot");
+		if (showAnonymous)
+			requestBuilder.param("wlshow", "anon");
+		else
+			requestBuilder.param("wlshow", "!anon");
+		if (showMinor)
+			requestBuilder.param("wlshow", "minor");
+		else
+			requestBuilder.param("wlshow", "!minor");
+		if (limit < 1)
+			requestBuilder.param("wllimit", "max");
+		else
+			requestBuilder.param("wllimit", limit);
+		if (hasNextPageInfo()) {
+			requestBuilder.param("wlcontinue", getNextPageInfo());
+		}
+		log.debug("using query {}", requestBuilder.build());
+		return requestBuilder.buildGet();
+
+	}
+
+	@Override
+	protected ImmutableList<WatchResponse> parseElements(String json) {
+		try {
+			this.responseList = objectMapper.readValue(json,
+					WatchListResults.class);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return ImmutableList.copyOf(responseList.getResults());
+	}
+
+	@Override
+	protected Optional<String> parseHasMore(String s) {
+		if (responseList.canContinue()) {
+			return Optional.of(responseList.getContinueToken());
+		} else {
+			return Optional.absent();
+		}
+	}
+
+	public static Builder from(MediaWikiBot bot) {
+		return new Builder(bot);
+	}
+
+	public static class Builder {
+
+		private MediaWikiBot bot;
+		private int limit = -1;
+		private WatchListProperties[] properties;
+		private boolean showBots = true;
+		private boolean showAnonymous = true;
+		private int[] namespaces = new int[] { DEFAULT_NS };
+		private boolean showMinor;
+
+		public Builder(MediaWikiBot bot) {
+			this.bot = bot;
+		}
+		/**
+		 * How many results to return per request.
+		 * Use a negative value to return the maximum.
+		 * @param limit
+		 * @return
+		 */
+		public Builder withLimit(int limit) {
+			this.limit = limit;
+			return this;
+		}
+
+		/**
+		 * Only list pages in these namespaces
+		 * @param namespaces
+		 * @return
+		 */
+		public Builder withNamespaces(int... namespaces) {
+			this.namespaces = namespaces;
+			return this;
+		}
+
+		public Builder withProperties(WatchListProperties... properties) {
+			this.properties = properties;
+			return this;
+		}
+
+		public Builder showBots(boolean show) {
+			this.showBots = show;
+			return this;
+		}
+
+		public Builder showAnonymous(boolean show) {
+			this.showAnonymous = show;
+			return this;
+		}
+
+		public Builder showMinor(boolean show) {
+			this.showMinor = show;
+			return this;
+		}
+
+		public WatchList build() {
+			return new WatchList(this);
+		}
+
+		public int[] getNamespaces() {
```

Comment:
we don't need this public getters in this inner class, the fields should be accessible directly for ```WatchList```